armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1174,18 +1175,24 @@ def buildRingSchedule(
         findAssembly
 
         """
+        maxRingInCore = self.r.core.getNumRings()
+        if dischargeRing > maxRingInCore:
+            runLog.warning(f"Discharge ring {dischargeRing} is outside the core (max {maxRingInCore}).")
+        if chargeRing > maxRingInCore:
+            runLog.warning(f"Charge ring {chargeRing} is outside the core (max {maxRingInCore}).")
+
         # process arguments
         if dischargeRing is None:
             # default to convergent
             dischargeRing = 1
         if chargeRing is None:
-            chargeRing = self.r.core.getNumRings()
+            chargeRing = maxRingInCore
 
-        if chargeRing > dischargeRing and not jumpRingTo:
+        if chargeRing > dischargeRing and jumpRingTo is None:
             jumpRingTo = 1
-        elif not jumpRingTo:
+        elif jumpRingTo is None:
```

Comment:
All this conditional logic is very hard to follow. Is there a way to clean this up so it's easier to understand and test?

Reply:
I think this line is fine.  But, for my preference, at least some comments on the if-statements, or something. Just to make them easier to understand.

Reply:
I added some comments